[Android] Cannot add task 'uploadSourcemaps' as a task with that name already exists.

Android build is failing due to duplicate tasks with same name, something is wrong with flavor naming on version `11.12.0`. `11.10.0` works fine.

### Steps to Reproduce the Problem

`./gradlew clean bundleRelease`

### Expected Behavior

Build compiles successfully. 

### Actual Behavior

Byu

```console
FAILURE: Build failed with an exception.
* Where:
Script '/Users/App.Release/builds/aknPzrPd/0/our-app/node_modules/instabug-reactnative/android/sourcemaps.gradle' line: 23
* What went wrong:
Cannot add task 'uploadSourcemaps' as a task with that name already exists.
```

### SDK Version
```json
    "instabug-reactnative": "^11.12.0",
```

### React Native, iOS and Android Versions

```json
    "react-native": "^0.71.7",
```

hi @a7medev, sure - here it goes:

```console
➜  android git:(cd437d0e) ✗ ./gradlew tasks
Configuration on demand is an incubating feature.

> Configure project :app
##############################

Using react.gradle in your project is deprecated! You should move to "apply plugin: com.facebook.react"
react.gradle is going to be removed in a future React Native project and your project will not build anymore.
You can use the template as a reference:
https://github.com/facebook/react-native/blob/main/template/android/app/build.gradle

##############################
WARNING:DSL element 'dexOptions' is obsolete and should be removed.
It will be removed in version 8.0 of the Android Gradle plugin.
Using it has no effect, and the AndroidGradle plugin optimizes dexing automatically.

> Configure project :react-native-blob-util
WARNING:Software Components will not be created automatically for Maven publishing from Android Gradle Plugin 8.0. To opt-in to the future behavior, set the Gradle property android.disableAutomaticComponentCreation=true in the `gradle.properties` file or use the new publishing DSL.

> Configure project :react-native-firebase_analytics
:react-native-firebase_analytics package.json found at /Users/lukebars/Desktop/product/x-app/node_modules/@react-native-firebase/analytics/package.json
:react-native-firebase_app package.json found at /Users/lukebars/Desktop/product/x-app/node_modules/@react-native-firebase/app/package.json
:react-native-firebase_analytics:firebase.bom using default value: 30.5.0
:react-native-firebase_analytics package.json found at /Users/lukebars/Desktop/product/x-app/node_modules/@react-native-firebase/analytics/package.json
:react-native-firebase_analytics:version set from package.json: 14.12.0 (14,12,0 - 14012000)
:react-native-firebase_analytics:android.compileSdk using custom value: 33
:react-native-firebase_analytics:android.targetSdk using custom value: 33
:react-native-firebase_analytics:android.minSdk using custom value: 23
:react-native-firebase_analytics:reactNativeAndroidDir /Users/lukebars/Desktop/product/x-app/node_modules/react-native/android

> Configure project :react-native-firebase_app
:react-native-firebase_app package.json found at /Users/lukebars/Desktop/product/x-app/node_modules/@react-native-firebase/app/package.json
:react-native-firebase_app:firebase.bom using default value: 30.5.0
:react-native-firebase_app:play.play-services-auth using default value: 20.3.0
:react-native-firebase_app package.json found at /Users/lukebars/Desktop/product/x-app/node_modules/@react-native-firebase/app/package.json
:react-native-firebase_app:version set from package.json: 14.12.0 (14,12,0 - 14012000)
:react-native-firebase_app:android.compileSdk using custom value: 33
:react-native-firebase_app:android.targetSdk using custom value: 33
:react-native-firebase_app:android.minSdk using custom value: 23
:react-native-firebase_app:reactNativeAndroidDir /Users/lukebars/Desktop/product/x-app/node_modules/react-native/android

> Configure project :react-native-firebase_auth
:react-native-firebase_auth package.json found at /Users/lukebars/Desktop/product/x-app/node_modules/@react-native-firebase/auth/package.json
:react-native-firebase_app package.json found at /Users/lukebars/Desktop/product/x-app/node_modules/@react-native-firebase/app/package.json
:react-native-firebase_auth:firebase.bom using default value: 30.5.0
:react-native-firebase_auth package.json found at /Users/lukebars/Desktop/product/x-app/node_modules/@react-native-firebase/auth/package.json
:react-native-firebase_auth:version set from package.json: 14.12.0 (14,12,0 - 14012000)
:react-native-firebase_auth:android.compileSdk using custom value: 33
:react-native-firebase_auth:android.targetSdk using custom value: 33
:react-native-firebase_auth:android.minSdk using custom value: 23
:react-native-firebase_auth:reactNativeAndroidDir /Users/lukebars/Desktop/product/x-app/node_modules/react-native/android

> Configure project :react-native-firebase_crashlytics
:react-native-firebase_crashlytics package.json found at /Users/lukebars/Desktop/product/x-app/node_modules/@react-native-firebase/crashlytics/package.json
:react-native-firebase_app package.json found at /Users/lukebars/Desktop/product/x-app/node_modules/@react-native-firebase/app/package.json
:react-native-firebase_crashlytics:firebase.bom using default value: 30.5.0
:react-native-firebase_crashlytics package.json found at /Users/lukebars/Desktop/product/x-app/node_modules/@react-native-firebase/crashlytics/package.json
:react-native-firebase_crashlytics:version set from package.json: 14.12.0 (14,12,0 - 14012000)
:react-native-firebase_crashlytics:android.compileSdk using custom value: 33
:react-native-firebase_crashlytics:android.targetSdk using custom value: 33
:react-native-firebase_crashlytics:android.minSdk using custom value: 23
:react-native-firebase_crashlytics:reactNativeAndroidDir /Users/lukebars/Desktop/product/x-app/node_modules/react-native/android

> Configure project :react-native-firebase_database
:react-native-firebase_database package.json found at /Users/lukebars/Desktop/product/x-app/node_modules/@react-native-firebase/database/package.json
:react-native-firebase_app package.json found at /Users/lukebars/Desktop/product/x-app/node_modules/@react-native-firebase/app/package.json
:react-native-firebase_database:firebase.bom using default value: 30.5.0
:react-native-firebase_database package.json found at /Users/lukebars/Desktop/product/x-app/node_modules/@react-native-firebase/database/package.json
:react-native-firebase_database:version set from package.json: 14.12.0 (14,12,0 - 14012000)
:react-native-firebase_database:android.compileSdk using custom value: 33
:react-native-firebase_database:android.targetSdk using custom value: 33
:react-native-firebase_database:android.minSdk using custom value: 23
:react-native-firebase_database:reactNativeAndroidDir /Users/lukebars/Desktop/product/x-app/node_modules/react-native/android

> Configure project :react-native-firebase_remote-config
:react-native-firebase_remote-config package.json found at /Users/lukebars/Desktop/product/x-app/node_modules/@react-native-firebase/remote-config/package.json
:react-native-firebase_app package.json found at /Users/lukebars/Desktop/product/x-app/node_modules/@react-native-firebase/app/package.json
:react-native-firebase_remote-config:firebase.bom using default value: 30.5.0
:react-native-firebase_remote-config package.json found at /Users/lukebars/Desktop/product/x-app/node_modules/@react-native-firebase/remote-config/package.json
:react-native-firebase_remote-config:version set from package.json: 14.12.0 (14,12,0 - 14012000)
:react-native-firebase_remote-config:android.compileSdk using custom value: 33
:react-native-firebase_remote-config:android.targetSdk using custom value: 33
:react-native-firebase_remote-config:android.minSdk using custom value: 23
:react-native-firebase_remote-config:reactNativeAndroidDir /Users/lukebars/Desktop/product/x-app/node_modules/react-native/android

> Configure project :react-native-flipper
WARNING:Software Components will not be created automatically for Maven publishing from Android Gradle Plugin 8.0. To opt-in to the future behavior, set the Gradle property android.disableAutomaticComponentCreation=true in the `gradle.properties` file or use the new publishing DSL.

> Configure project :react-native-reanimated
AAR for react-native-reanimated has been found
/Users/lukebars/Desktop/product/x-app/node_modules/react-native-reanimated/android/react-native-reanimated-71-hermes.aar

FAILURE: Build failed with an exception.

* Where:
Script '/Users/lukebars/Desktop/product/x-app/node_modules/instabug-reactnative/android/sourcemaps.gradle' line: 23

* What went wrong:
Cannot add task 'uploadSourcemaps' as a task with that name already exists.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.5.1/userguide/command_line_interface.html#sec:command_line_warnings
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,11 @@
 # Changelog
 
+## [Unreleased](https://github.com/Instabug/Instabug-React-Native/compare/v11.12.0...dev)
+
+### Fixed
+
+- Fix an issue with the Android sourcemaps upload Gradle task getting recreated when both `bundleReleaseJsAndAssets` and `createBundleReleaseJsAndAssets` tasks exist in the same Android project ([#991](https://github.com/Instabug/Instabug-React-Native/pull/991)), closes [#989](https://github.com/Instabug/Instabug-React-Native/issues/989).
+
 ## [11.12.0](https://github.com/Instabug/Instabug-React-Native/compare/v11.10.0...11.12.0) (May 30, 2023)
 
 ### Changed

--- a/android/sourcemaps.gradle
+++ b/android/sourcemaps.gradle
@@ -21,9 +21,9 @@ gradle.projectsEvaluated {
 Task createUploadSourcemapsTask(String flavor) {
     def name = 'uploadSourcemaps' + flavor.capitalize()
 
-    // Don't recreate the task if it already exsts.
+    // Don't recreate the task if it already exists.
     // This prevents the build from failing in an edge case where the user has
-    // both `bundleReleaseJsAndAssets` and `createBundleJsAndAssets`
+    // both `bundleReleaseJsAndAssets` and `createBundleReleaseJsAndAssets`
     def taskExists = tasks.getNames().contains(name)
     if (taskExists) {
         return tasks.named(name).get()

--- a/android/sourcemaps.gradle
+++ b/android/sourcemaps.gradle
@@ -20,6 +20,15 @@ gradle.projectsEvaluated {
 
 Task createUploadSourcemapsTask(String flavor) {
     def name = 'uploadSourcemaps' + flavor.capitalize()
+
+    // Don't recreate the task if it already exsts.
+    // This prevents the build from failing in an edge case where the user has
+    // both `bundleReleaseJsAndAssets` and `createBundleJsAndAssets`
+    def taskExists = tasks.getNames().contains(name)
+    if (taskExists) {
+        return tasks.named(name).get()
+    }
+
     def provider = tasks.register(name) {
         group 'instabug'
         description 'Uploads sourcemaps file to Instabug server'

